vm/src/compiler.rs

```diff
@@ -226,38 +213,40 @@ impl CompilerEnv for TypeInfos {
     }
 }
 
-pub struct Compiler<'a> {
-    globals: &'a (CompilerEnv + 'a),
+pub struct Compiler<'a, E> {
```

Comment:
I guess one could implement `impl CompilerEnv for [T]` so `&self: &[T]` which cannot be casted to `&TypeEnv` (as that would require it to return `(v-table, size, data-ptr)` or something)